docs/docker_install.md

```diff
@@ -0,0 +1,150 @@
+request as it is easiest way to robustly check your commits. Doing so
+doesn't require any deeper understanding of docker or docker-compose
+and should run out-of the box.
+
+By default it will build and run tests only in SGX simulator mode but
+the makefile does honor
+the `SGX_MODE` environment variable if you want to test in SGX HW
+mode (in which case, though, you will have to put proper sgx attestation info
+files (`sgx_spid.txt`,  `sgx_spid_api_key`, `sgx_ias_key.pem`) into the
+sgx sub-directory or define the `PDO_SGX_KEY_ROOT` environment variable to
+point to a suiteable destination.
+See `../build/common-config.sh --help` for more information on
+`PDO_SGX_KEY_ROOT` and related settings.
+
+
+## Developing and manual testing with docker(-compose)
+
+If you want to run your own experiments in the test environment, you
+can run
+ ```bash
+make test-env-setup
+```
+which launches the various containers of the topology. You can then
+interact with the environment using the standard docker-compose commands like
+`exec` to log into individual containers, `ps` to see status, `top` to monitor
+cpu inside containers or `down` to shut the invironment down again.
+See [docker-compose commandline reference](https://docs.docker.com/compose/reference/)
+for more information on commands.  To make sure these commands run
+with settings consistent with on how you invoked `test-env-setup`, you
+can invoke them with also with the makefile target  `run` . Just
+invoke it as
+```bash
+	make run ARGS='<docker-compose subcommand & args>'
+```
+e.g.  run `make run ARGS='exec pdo-build bash'` to start a shell in
+the pdo-build container.
+
+If you define the `PDO_DEBUG_BUILD` environment variable, the make
+commands will (with help of [`sawtooth.debugging.yaml`](sawtooth.debugging.yaml)) build
+the code with debugging and and run docker containers such that
+gdb/sgx-gdb-based debugging is possible.
+Note though, that due to some docker(-compose)ism, terminating daemon
+processes such as the ones started
+by ps-start/es-start will run in zombie processes. They don't hold any
+resources such as sockets or alike and subseequent
+\*-start/\*-status/\*-stop script should work as expected.
+
+While you can run end-to-end tests inside docker, sometimes it might
+be easier to test outside so you can, e.g., monitor localhost traffic
+with `wireshark` which is challenging within docker. Note that with
+`make test-env-setup` (or `make test-env-setup-with-no-build` if you
+are sure container images for PDO-TP and other components are already
+properly built) you get a fresh sawtooth setup where ledger rest API
+is also exposed to the host, i.e., the default url
+`http://localhost:8008` does also work from the host and you can test
+client and {e,s,p}services on the host with a fresh and
+self-contained/single machine installation.
+
+For more advanced docker-compose usage, check the headers in the yaml
+files:
```

Comment:
all links in these two lists are broken, see above

Reply:
see above